src/fs.rs

```diff
@@ -161,18 +161,22 @@ impl BiosParameterBlock {
 
         // bytes per sector is u16, sectors per cluster is u8, so guaranteed no overflow in multiplication
         let bytes_per_cluster = bpb.bytes_per_sector as u32 * bpb.sectors_per_cluster as u32;
-        let maximum_compatibility_bytes_per_cluster : u32 = 32 * 1024;
+        let maximum_compatibility_bytes_per_cluster : u32 = match bpb.bytes_per_sector {
```

Comment:
The warning is not because rust-fatfs does not handle the situation.  The warning is to warn of potential incompatibility when the media is moved to other systems.  (e.g., dvd players, digital cameras, security cameras, drones, IoT devices, etc.)

Allow me to describe it differently.  Legacy FAT code existed when the only sector size supported was 512 bytes per sector.  Thus, for media that is 512 bytes per sector, it's not known how old the file system implementation is which will eventually read the media.  FAT is useful because it's compatible, even on these really old FAT implementations.

Then, 4k logical sector size disks started shipping.  Legacy FAT solutions that cause the 32k limit requirement don't support 4k logical sector size.  This frees the FAT file system, allowing 64k and larger cluster sizes.  But, to be safe, because 4k == 512 * 8, they limit to 32k * 8 (256k), as the number of sectors per cluster remains in the same range, minimizing overall risk while allowing for greater cluster sizes.  In other words, Microsoft made a calculated risk to allow increased cluster size for 4k logical sector size media.

For any disk using 512 bytes per sector, the limit remains 32k for highest compatibility.
However, for larger logical sector size media, the limit has been officially raised to 256k clusters, while maintaining similarly high compatibility (if the device supports 4k sectors).

I recommend keeping the more specific warning, to warn that such a file system does not have the same levels of compatibility.

Reply:
"Legacy FAT solutions that cause the 32k limit requirement don't support 4k logical sector size." I don't agree with that. sectors_per_cluster field can be set to value 1, 2, 4 and 8 and bytes_per_cluster would still met limit of 32K. How 4k sectors require you to have clusters bigger than 32K? Cluster size of 32K allows partition size of 8TB so I don't see a reason why would anybody use bigger cluster size. Still I can't find any source of information which would say 256K cluster size limit should be used in such case. format.exe output in Windows 10 is not a specification.
This warning says about incompatibility with some implementations and 256K cluser size is definitely incompatible with some systems like Windows 98 so I don't think using it harms anybody.

Reply:
Hi Rafal,
I disagree.  Even so, I have removed this change.  See 1512736a3a662f86a28bfe8615f351a52c6e677e